src/main/scala/scala/collection/decorators/SeqDecorator.scala

```diff
@@ -41,4 +41,49 @@ class SeqDecorator[C, S <: HasSeqOps[C]](coll: C)(implicit val seq: S) {
   def intersperse[B >: seq.A, That](start: B, sep: B, end: B)(implicit bf: BuildFrom[C, B, That]): That =
     bf.fromSpecificIterable(coll)(new View.IntersperseSurround(seq(coll), start, sep, end))
 
+  /** Splits this collection into groups according to the given predicate.
+    *
+    * @param p the predicate used to discriminate elements
+    * @return A nested collection with groups of elements,
+    *         opening new groups whenever the predicate
+    *         changes the return type
+    *
+    * @example {{{
+    * // Example 1: Split a list of integers into groups that are even / odd
+    * List(1, 2, 4, 6, 7).splitWith(i => i % 2 == 0) => List(List(1), List(2, 4, 6), List(7))
+    *
+    * // Example 2: Split a list of chars into groups that are upper case or lower case
+    * List('a', 'b', 'C', 'D', 'e', 'f').splitWith(_.isUpper) => List(List('a', 'b'), List('C', 'D'), List('e', 'f'))
+    * }}}
+    */
+  def splitWith[That](p: seq.A => Boolean)(implicit bf: BuildFrom[C, seq.A, That]): That = {
+    def newGroupBuilder() = seq().iterableFactory.newBuilder[seq.A]
+
+    //TODO: How to get a builder of a nested target collection?
```

Comment:
You need to take two `BuildFrom` parameters:

~~~ scala
def splitWith[Group, That](p: seq.A => Boolean)(implicit group: BuildFrom[C, seq.A, Group], bf: BuildFrom[C, Group, That]): That
~~~

Reply:
Ah I see, thanks!